BUILD.bazel

```diff
@@ -1671,19 +1672,76 @@ cc_library(
 
 cc_library(
     name = "torch_python",
-    srcs = libtorch_python_core_sources + GENERATED_AUTOGRAD_PYTHON,
+    srcs = libtorch_python_core_sources
+        + if_cuda(libtorch_python_cuda_sources)
+        + if_cuda(libtorch_python_distributed_sources)
+        + GENERATED_AUTOGRAD_PYTHON,
+    hdrs = glob([
+        "torch/csrc/generic/*.cpp",
+    ]),
+    copts = COMMON_COPTS + if_cuda(["-DUSE_CUDA=1"]),
     deps = [
-        ":shm",
         ":torch",
+        ":shm",
         "@pybind11",
     ],
 )
 
 pybind_extension(
-    name = "_C",
+    name = "torch/_C",
     srcs = ["torch/csrc/stub.c"],
     deps = [
         ":torch_python",
+        ":aten_nvrtc",
+    ],
+)
+
+cc_binary(
+    name = "torch/bin/torch_shm_manager",
+    srcs = [
+        "torch/lib/libshm/manager.cpp",
+    ],
+    deps = [
+        ":shm",
+    ],
+    linkstatic = False,
+)
+
+template_rule(
+    name = "gen_version_py",
+    src = ":torch/version.py.tpl",
+    out = "torch/version.py",
+    substitutions = if_cuda({
+        # Set default to 11.2. Otherwise Torchvision complains about incompatibility.
+        "{{CUDA_VERSION}}": "11.2",
+        "{{VERSION}}": "2.0.0",
+        # TODO: can we automatically get correct git SHA here?
+        "{{GIT_VERSION}}": "0000000000000000000000000000000000000000",
```

Comment:
have you seen https://bazel.build/docs/user-manual#workspace-status?

ideally we only "stamp" if it's been requested.